src/ripple/app/paths/tests/Path_test.cpp

```diff
@@ -1,212 +1,585 @@
+  ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+  OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 */
 //==============================================================================
 
-#include <ripple/app/ledger/tests/common_ledger.h>
-#include <ripple/crypto/KeyType.h>
-#include <ripple/json/json_writer.h>
-#include <ripple/basics/TestSuite.h>
+#include <BeastConfig.h>
+#include <ripple/test/jtx.h>
+#include <ripple/json/to_string.h>
+#include <ripple/protocol/TxFlags.h>
+#include <beast/unit_test/suite.h>
 
-// probably going to be moved to common_ledger.h
+#include <ripple/rpc/impl/RipplePathFind.h>
+#include <ripple/json/json_reader.h>
+#include <ripple/basics/Log.h>
 
 namespace ripple {
 namespace test {
 
-class Path_test : public TestSuite
+using namespace jtx;
+
+class Path_test : public beast::unit_test::suite
 {
+public:
+    Json::Value
+    findPath (std::shared_ptr<Ledger> ledger, Account const& src,
+        Account const& dest, std::vector<STAmount> srcCurrencies,
+            STAmount const& saDstAmount)
+    {
+        auto jvSrcCurrencies = Json::Value(Json::arrayValue);
+        for(auto const& srcCurrency : srcCurrencies)
+            jvSrcCurrencies.append(srcCurrency.getJson(0));
+
+        boost::optional<Json::Value> contextPaths;
+        int const level = 8;
+        auto result = ripplePathFind(
+            std::make_shared<RippleLineCache>(
+                ledger), src.pk(), dest.pk(), saDstAmount,
+            ledger, jvSrcCurrencies, contextPaths, level);
+        if(!result.first)
+            throw std::runtime_error(
+            "Path_test::findPath: ripplePathFind find failed");
+
+        return result.second;
+    }
+
     void
     test_no_direct_path_no_intermediary_no_alternatives()
     {
         testcase("no direct path no intermediary no alternatives");
-        std::uint64_t const xrp = std::mega::num;
-
-        auto master = createAccount("masterpassphrase", KeyType::ed25519);
-
-        std::shared_ptr<Ledger const> LCL;
-        Ledger::pointer ledger;
-        std::tie(LCL, ledger) = createGenesisLedger(100000 * xrp, master);
-
-        auto accounts = createAndFundAccounts(master, { "alice", "bob" }, 
-            KeyType::ed25519, 10000 * xrp, ledger);
-        auto& alice = accounts["alice"];
-        auto& bob = accounts["bob"];
-        expectNotEquals(alice.pk.humanAccountID(), master.pk.humanAccountID());
-        
-        auto alternatives = findPath(ledger, alice, bob, { Currency("USD") },
-            Amount(5, "USD", alice), log.stream());
-        log << "ripplePathFind alternatives: " << alternatives;
+        Env env(*this);
+        env.fund(XRP(10000), "alice", "bob");
 
-        expectEquals(alternatives.size(), 0);
+        auto const alternatives = findPath(env.ledger, "alice", "bob",
+            {Account("alice")["USD"](0)}, Account("bob")["USD"](5));
+        expect(alternatives.size() == 0);
```

Comment:
Nice to compare old and new version and see how much cleaner this test is with the new jtx framework. Such a win.